packages/common/src/utils/filters.ts

```diff
@@ -376,3 +379,103 @@ export const getFilterRulesByFieldType = (
             metrics: [],
         },
     );
+
+export const getDashboardFilterRulesForTile = (
+    tileUuid: string,
+    tables: string[],
+    rules: DashboardFilterRule[],
+): DashboardFilterRule[] =>
+    rules
+        .filter((rule) => !rule.disabled)
+        .map((filter) => {
+            const tileConfig = filter.tileTargets?.[tileUuid];
+
+            // If the config is false, we remove this filter
+            if (tileConfig === false) {
+                return null;
+            }
+
+            // If the tile isn't in the tileTarget overrides,
+            // we return the filter and don't treat this tile
+            // differently.
+            if (tileConfig === undefined) {
+                return filter;
+            }
+
+            return {
+                ...filter,
+                target: {
+                    fieldId: tileConfig.fieldId,
+                    tableName: tileConfig.tableName,
+                },
+            };
+        })
+        .filter((f): f is DashboardFilterRule => f !== null)
+        .filter((f) => tables.includes(f.target.tableName));
+
+export const getDashboardFiltersForTile = (
+    tileUuid: string,
+    tables: string[],
+    dashboardFilters: DashboardFilters,
+): DashboardFilters => ({
+    dimensions: getDashboardFilterRulesForTile(
+        tileUuid,
+        tables,
+        dashboardFilters.dimensions,
+    ),
+    metrics: getDashboardFilterRulesForTile(
+        tileUuid,
+        tables,
+        dashboardFilters.metrics,
+    ),
+});
+
+export const addDashboardFiltersToMetricQuery = (
+    explore: Explore,
+    metricQuery: MetricQuery,
+    tileUuid: string | undefined,
+    dashboardFilters: DashboardFilters | undefined,
+): MetricQuery => {
+    const tables = Object.keys(explore.tables);
+
+    const dashboardFiltersForTile =
+        tileUuid && dashboardFilters
+            ? getDashboardFiltersForTile(tileUuid, tables, dashboardFilters)
+            : undefined;
+
+    const dimensionFilters: FilterGroup | undefined =
+        dashboardFiltersForTile && dashboardFiltersForTile.dimensions.length > 0
+            ? {
+                  id: 'yes',
+                  and: [
+                      ...(metricQuery.filters.dimensions
+                          ? [metricQuery.filters.dimensions]
+                          : []),
+                      ...dashboardFiltersForTile.dimensions,
+                  ],
+              }
+            : metricQuery.filters.dimensions;
+
+    const metricFilters: FilterGroup | undefined =
+        dashboardFiltersForTile && dashboardFiltersForTile.metrics.length > 0
+            ? {
+                  id: 'no',
+                  and: [
+                      ...(metricQuery.filters.metrics
+                          ? [metricQuery.filters.metrics]
+                          : []),
+                      ...dashboardFiltersForTile.metrics,
+                  ],
+              }
```

Comment:
hey @ZeRego is this the right way to combine filters? in this case dashboard filters to the existing metric filters? if yes, does it matter what the id's will be? I copy pasted this from here: https://github.com/jayanand05/lightdash/blob/360ebbcaf1b74b49ff982193d036f1b4e149bff0/packages/frontend/src/hooks/dashboard/useSavedQueryWithDashboardFilters.ts#L62

I was also thinking to extract this as a function, and name it something like, "combineFilters"

Reply:
> is this the right way to combine filters?

Yes that is correct. 
We also have a static method `ProjectService.combineFilters` that does almost the same.

> does it matter what the id's will be?

not at the moment

Reply:
@ZeRego great, as this isn't going to hit the DB and the feature's working - I'm gonna merge it. [Created a ticket](https://github.com/lightdash/lightdash/issues/7246) to tackle duplicated code tomorrow. which will combine `combineFilters` function for both frontend and backend 👌